OpenAPI generator should support fully qualified class names

When running the OpenAPI generator I want it to support fully qualified class names so that I can run the generator on a method like `public java.time.Instant getFqdnInstant() {}`.

Current behavior: an `java.lang.RuntimeException: Unable to load RELATIVE ref: ./java.time.Instant` exception is thrown from the `generate-connect-modules` maven task.

Source:
```java
@VaadinService
public class DemoVaadinService {

  // does not work
  public java.time.Instant getFqdnInstant() {
    return Instant.now();
  }

  // works fine
  public Instant getInstant() {
    return Instant.now();
  }
}
```

<details>
  <summary>Exception stack trace</summary>
<pre>
[WARNING] Exception while reading:
java.lang.RuntimeException: Unable to load RELATIVE ref: ./java.time.Instant path: /Users/viktor/IdeaProjects/vaadin-services/vaadin-connect/vaadin-connect-demo/target/generated-resources
    at io.swagger.v3.parser.util.RefUtils.readExternalRef (RefUtils.java:183)
    at io.swagger.v3.parser.ResolverCache.loadRef (ResolverCache.java:118)
    at io.swagger.v3.parser.processors.ExternalRefProcessor.processRefToExternalSchema (ExternalRefProcessor.java:49)
    at io.swagger.v3.parser.processors.SchemaProcessor.processReferenceSchema (SchemaProcessor.java:201)
    at io.swagger.v3.parser.processors.SchemaProcessor.processSchema (SchemaProcessor.java:36)
    at io.swagger.v3.parser.processors.ResponseProcessor.processResponse (ResponseProcessor.java:52)
    at io.swagger.v3.parser.processors.OperationProcessor.processOperation (OperationProcessor.java:67)
    at io.swagger.v3.parser.processors.PathsProcessor.processPaths (PathsProcessor.java:84)
    at io.swagger.v3.parser.OpenAPIResolver.resolve (OpenAPIResolver.java:49)
    at io.swagger.v3.parser.OpenAPIV3Parser.readLocation (OpenAPIV3Parser.java:53)
    at io.swagger.parser.OpenAPIParser.readLocation (OpenAPIParser.java:19)
    at io.swagger.codegen.v3.config.CodegenConfigurator.toClientOptInput (CodegenConfigurator.java:492)
    at com.vaadin.connect.plugin.generator.VaadinConnectJsGenerator.launch (VaadinConnectJsGenerator.java:179)
    at com.vaadin.connect.plugin.EsModuleGeneratorMojo.execute (EsModuleGeneratorMojo.java:39)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.RuntimeException: Could not find ./java.time.Instant on the classpath
    at io.swagger.v3.parser.util.ClasspathHelper.loadFileFromClasspath (ClasspathHelper.java:31)
    at io.swagger.v3.parser.util.RefUtils.readExternalRef (RefUtils.java:177)
    at io.swagger.v3.parser.ResolverCache.loadRef (ResolverCache.java:118)
    at io.swagger.v3.parser.processors.ExternalRefProcessor.processRefToExternalSchema (ExternalRefProcessor.java:49)
    at io.swagger.v3.parser.processors.SchemaProcessor.processReferenceSchema (SchemaProcessor.java:201)
    at io.swagger.v3.parser.processors.SchemaProcessor.processSchema (SchemaProcessor.java:36)
    at io.swagger.v3.parser.processors.ResponseProcessor.processResponse (ResponseProcessor.java:52)
    at io.swagger.v3.parser.processors.OperationProcessor.processOperation (OperationProcessor.java:67)
    at io.swagger.v3.parser.processors.PathsProcessor.processPaths (PathsProcessor.java:84)
    at io.swagger.v3.parser.OpenAPIResolver.resolve (OpenAPIResolver.java:49)
    at io.swagger.v3.parser.OpenAPIV3Parser.readLocation (OpenAPIV3Parser.java:53)
    at io.swagger.parser.OpenAPIParser.readLocation (OpenAPIParser.java:19)
    at io.swagger.codegen.v3.config.CodegenConfigurator.toClientOptInput (CodegenConfigurator.java:492)
    at com.vaadin.connect.plugin.generator.VaadinConnectJsGenerator.launch (VaadinConnectJsGenerator.java:179)
    at com.vaadin.connect.plugin.EsModuleGeneratorMojo.execute (EsModuleGeneratorMojo.java:39)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:134)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:208)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:154)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:146)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:51)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:309)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:194)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:107)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:955)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:290)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:194)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
</pre>
</details>

--- a/vaadin-connect-maven-plugin/src/main/java/com/vaadin/connect/plugin/generator/OpenApiParser.java
+++ b/vaadin-connect-maven-plugin/src/main/java/com/vaadin/connect/plugin/generator/OpenApiParser.java
@@ -18,7 +18,6 @@ package com.vaadin.connect.plugin.generator;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
@@ -26,8 +25,6 @@ import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 import com.github.javaparser.ParseResult;
 import com.github.javaparser.ast.CompilationUnit;
@@ -79,10 +76,6 @@ class OpenApiParser {
   private static final List<String> COLLECTION_TYPES = Arrays.asList(
       "collection", "list", "arraylist", "linkedlist", "set", "hashset",
       "sortedset", "treeset");
-  private static final List<String> PREDEFINED_TYPES = Stream
-      .of(NUMBER_TYPES, STRING_TYPES, BOOLEAN_TYPES, MAP_TYPES,
-          COLLECTION_TYPES)
-      .flatMap(Collection::stream).collect(Collectors.toList());
   private List<Path> javaSourcePaths = new ArrayList<>();
   private OpenApiConfiguration configuration;
   private Set<String> usedSchemas;
@@ -219,29 +212,13 @@ class OpenApiParser {
       if (field.isTransient()) {
         continue;
       }
-      field.getVariables().forEach(variableDeclarator -> {
-        if (isBeanType(variableDeclarator.getType())) {
-          Schema propertyItem = new Schema()
-              .$ref(variableDeclarator.getTypeAsString());
-          schema.addProperties(variableDeclarator.getNameAsString(),
-              propertyItem);
-          usedSchemas.add(variableDeclarator.getTypeAsString());
-        } else if (isMapType(variableDeclarator.getType())) {
-          schema.addProperties(variableDeclarator.getNameAsString(),
-              parseTypeToSchema(variableDeclarator.getType()));
-        } else {
-          Schema propertyItem = parseTypeToSchema(variableDeclarator.getType());
-          schema.addProperties(variableDeclarator.getNameAsString(),
-              propertyItem);
-        }
-      });
+      field.getVariables()
+          .forEach(variableDeclarator -> schema.addProperties(
+              variableDeclarator.getNameAsString(),
+              parseTypeToSchema(variableDeclarator.getType())));
     }
     parseInnerClasses(typeDeclaration.asClassOrInterfaceDeclaration());
     return schema;
-  }
-
-  private boolean isMapType(Type type) {
-    return MAP_TYPES.contains(getTypeName(type));
   }
 
   private Map<String, PathItem> createPathItems(
@@ -317,15 +294,8 @@ class OpenApiParser {
 
   private MediaType createReturnMediaType(MethodDeclaration methodDeclaration) {
     MediaType mediaItem = new MediaType();
-    Schema schema;
-    if (isBeanType(methodDeclaration.getType())) {
-      String type = methodDeclaration.getType().asString();
-      schema = new Schema().$ref(type);
-      usedSchemas.add(type);
-    } else {
-      schema = parseTypeToSchema(methodDeclaration.getType());
-    }
-    mediaItem.schema(schema);
+    Type methodReturnType = methodDeclaration.getType();
+    mediaItem.schema(parseTypeToSchema(methodReturnType));
     return mediaItem;
   }
 
@@ -348,14 +318,7 @@ class OpenApiParser {
     Schema requestSchema = new ObjectSchema();
     requestBodyObject.schema(requestSchema);
     methodDeclaration.getParameters().forEach(parameter -> {
-      Schema paramSchema;
-      if (isBeanType(parameter.getType())) {
-        paramSchema = new Schema().$ref(parameter.getTypeAsString());
-        usedSchemas.add(parameter.getTypeAsString());
-      } else {
-        parameter.getType().getElementType();
-        paramSchema = parseTypeToSchema(parameter.getType());
-      }
+      Schema paramSchema = parseTypeToSchema(parameter.getType());
       paramSchema
           .description(paramsDescription.get(parameter.getNameAsString()));
       requestSchema.addProperties(parameter.getNameAsString(), paramSchema);
@@ -363,20 +326,11 @@ class OpenApiParser {
     return requestBody;
   }
 
-  private boolean isBeanType(Type type) {
-    if (type.isPrimitiveType()) {
-      return false;
-    }
-    String typeName = getTypeName(type);
-    boolean isPredefinedType = PREDEFINED_TYPES.contains(typeName);
-    return !type.isArrayType() && type.isReferenceType() && !isPredefinedType;
-  }
-
   private Schema parseTypeToSchema(Type javaType) {
     if (javaType.isArrayType()) {
       return createArraySchema(javaType);
     }
-    String typeName = getTypeName(javaType);
+    String typeName = getTypeName(javaType).toLowerCase(Locale.ENGLISH);
     if (NUMBER_TYPES.contains(typeName)) {
       return new NumberSchema();
     } else if (STRING_TYPES.contains(typeName)) {
@@ -392,7 +346,7 @@ class OpenApiParser {
   }
 
   private Schema createUserBeanSchema(Type javaType) {
-    String userType = javaType.asString();
+    String userType = getTypeName(javaType);
     usedSchemas.add(userType);
     return new ObjectSchema().$ref(userType);
   }
@@ -400,10 +354,9 @@ class OpenApiParser {
   private String getTypeName(Type javaType) {
     String typeName;
     if (javaType.isClassOrInterfaceType()) {
-      typeName = javaType.asClassOrInterfaceType().getNameAsString()
-          .toLowerCase(Locale.ENGLISH);
+      typeName = javaType.asClassOrInterfaceType().getNameAsString();
     } else {
-      typeName = javaType.asString().toLowerCase(Locale.ENGLISH);
+      typeName = javaType.asString();
     }
     return typeName;
   }

--- a/vaadin-connect-maven-plugin/src/test/java/com/vaadin/connect/plugin/generator/GeneratorTestClass.java
+++ b/vaadin-connect-maven-plugin/src/test/java/com/vaadin/connect/plugin/generator/GeneratorTestClass.java
@@ -15,6 +15,7 @@
  */
 package com.vaadin.connect.plugin.generator;
 
+import java.time.Instant;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
@@ -107,6 +108,15 @@ public class GeneratorTestClass {
    */
   public boolean getTwoParameters(String input, int secondInput) {
     return false;
+  }
+
+  /**
+   * Get instant nano
+   * @param input input parameter
+   * @return current time as an Instant
+   */
+  public java.time.Instant fullFQNMethod(java.lang.Integer input) {
+    return Instant.now();
   }
 
   protected void hiddenMethod() {

--- a/vaadin-connect-maven-plugin/src/test/resources/com/vaadin/connect/plugin/generator/expected-GeneratorTestClass.js
+++ b/vaadin-connect-maven-plugin/src/test/resources/com/vaadin/connect/plugin/generator/expected-GeneratorTestClass.js
@@ -24,6 +24,16 @@ export class GeneratorTestClass {
   }
 
   /**
+   * Get instant nano
+   *
+   * @param {number} input - input parameter
+   * @returns {Promise<Instant>} Return current time as an Instant
+   */
+  fullFQNMethod(input) {
+    return this._client.call('GeneratorTestClass', 'fullFQNMethod', {input});
+  }
+
+  /**
    * Get the map of user and roles
    *
    * @returns {Promise<object>} Return map of user and roles
@@ -96,6 +106,7 @@ export class GeneratorTestClass {
 const service = new GeneratorTestClass(defaultClient);
 
 export const countUser = service.countUser.bind(service);
+export const fullFQNMethod = service.fullFQNMethod.bind(service);
 export const getAllUserRolesMap = service.getAllUserRolesMap.bind(service);
 export const getAllUsers = service.getAllUsers.bind(service);
 export const getArrayInt = service.getArrayInt.bind(service);

--- a/vaadin-connect-maven-plugin/src/test/resources/com/vaadin/connect/plugin/generator/expected-openapi-custom-application-properties.json
+++ b/vaadin-connect-maven-plugin/src/test/resources/com/vaadin/connect/plugin/generator/expected-openapi-custom-application-properties.json
@@ -143,6 +143,39 @@
         }
       }
     },
+    "/GeneratorTestClass/fullFQNMethod" : {
+      "post" : {
+        "tags" : [ "GeneratorTestClass" ],
+        "description" : "Get instant nano",
+        "requestBody" : {
+          "content" : {
+            "application/json" : {
+              "schema" : {
+                "type" : "object",
+                "properties" : {
+                  "input" : {
+                    "type" : "number",
+                    "description" : "input parameter"
+                  }
+                }
+              }
+            }
+          }
+        },
+        "responses" : {
+          "200" : {
+            "description" : "Return current time as an Instant",
+            "content" : {
+              "application/json" : {
+                "schema" : {
+                  "$ref" : "#/components/schemas/Instant"
+                }
+              }
+            }
+          }
+        }
+      }
+    },
     "/GeneratorTestClass/countUser" : {
       "post" : {
         "tags" : [ "GeneratorTestClass" ],

--- a/vaadin-connect-maven-plugin/src/test/resources/com/vaadin/connect/plugin/generator/expected-openapi.json
+++ b/vaadin-connect-maven-plugin/src/test/resources/com/vaadin/connect/plugin/generator/expected-openapi.json
@@ -143,6 +143,39 @@
         }
       }
     },
+    "/GeneratorTestClass/fullFQNMethod" : {
+      "post" : {
+        "tags" : [ "GeneratorTestClass" ],
+        "description" : "Get instant nano",
+        "requestBody" : {
+          "content" : {
+            "application/json" : {
+              "schema" : {
+                "type" : "object",
+                "properties" : {
+                  "input" : {
+                    "type" : "number",
+                    "description" : "input parameter"
+                  }
+                }
+              }
+            }
+          }
+        },
+        "responses" : {
+          "200" : {
+            "description" : "Return current time as an Instant",
+            "content" : {
+              "application/json" : {
+                "schema" : {
+                  "$ref" : "#/components/schemas/Instant"
+                }
+              }
+            }
+          }
+        }
+      }
+    },
     "/GeneratorTestClass/countUser" : {
       "post" : {
         "tags" : [ "GeneratorTestClass" ],

